system/core/base.php

```diff
@@ -41,7 +41,9 @@
 	include_once ('language.php');
 	include_once ('crypto.php');
 
-	if (!is_file('data/hypha.xml')) die('serious error: missing system file hypha.xml');
+	// jump to installation procedure when hypha.xml is missing
+	if (!is_file('data/hypha.xml')) header('Location: hypha.php');
+
```

Comment:
I've moved this check into index.php, so it can incorporate the root url. This fixes an infinite redirect when trying to access e.g. /en/SomePage without a datafile, which would redirect to /en/hypha.php, which would keep redirecting to itself.